service/frontend/adminHandler.go

```diff
@@ -421,6 +423,212 @@ func (adh *AdminHandler) GetWorkflowExecutionRawHistory(
 	return result, nil
 }
 
+// GetWorkflowExecutionRawHistoryV2 - retrieves the history of workflow execution
+func (adh *AdminHandler) GetWorkflowExecutionRawHistoryV2(
+	ctx context.Context,
+	request *admin.GetWorkflowExecutionRawHistoryRequestV2,
+) (resp *admin.GetWorkflowExecutionRawHistoryResponseV2, retError error) {
+
+	defer log.CapturePanic(adh.GetLogger(), &retError)
+	scope := metrics.AdminGetWorkflowExecutionRawHistoryV2Scope
+	sw := adh.startRequestProfile(scope)
+	defer sw.Stop()
+
+	if err := adh.validateGetWorkflowExecutionRawHistoryV2Request(
+		request,
+	); err != nil {
+		return nil, adh.error(err, scope)
+	}
+	domainID, err := adh.domainCache.GetDomainID(request.GetDomain())
+	if err != nil {
+		return nil, err
+	}
+
+	execution := request.Execution
+	response, err := adh.history.GetMutableState(ctx, &h.GetMutableStateRequest{
+		DomainUUID: common.StringPtr(domainID),
+		Execution:  execution,
+	})
+	if err != nil {
+		return nil, adh.error(err, scope)
+	}
+	nextEventID := request.GetNextEventId()
+	if nextEventID > response.GetNextEventId() {
+		nextEventID = response.GetNextEventId()
+	}
+	versionHistory, err := adh.getVersionHistory(
+		response.GetVersionHistories(),
+		nextEventID,
+		request.GetNextEventVersion())
+	if err != nil {
+		return nil, adh.error(err, scope)
+	}
+	token, err := adh.preparePaginationToken(
+		ctx,
+		request,
+		response.GetCurrentBranchToken(),
+		response.GetReplicationInfo(),
+		nextEventID)
+	if err != nil {
+		return nil, adh.error(err, scope)
+	}
+
+	var historyBatches []*gen.History
+	var size int
+	pageSize := int(request.GetMaximumPageSize())
+	shardID := common.WorkflowIDToHistoryShard(execution.GetWorkflowId(), adh.numberOfHistoryShards)
+	_, historyBatches, token.PersistenceToken, size, err = historyService.PaginateHistory(
```

Comment:
there is a new get raw history API from events store, plz use that

Reply:
Updated to use ReadRawHistoryBranch